template-parts/post-list-filters.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Template for post list filter controls
+ *
+ * @package shiro
+ */
+
+?>
+
+<section class="post-list-filter mw-980">
+
+	<div class="post-list-filter__head">
+
+		<h3>
+			<?php
+			$total_results = $wp_query->found_posts;
+			$paged = ( get_query_var( 'paged' ) ) ? get_query_var( 'paged' ) : 1;
+			$posts_per_page = get_query_var( 'posts_per_page' );
+			$first_result = ( $posts_per_page * $paged ) - $posts_per_page + 1;
+			$last_result = min( $total_results, $wp_query->post_count * $paged );
+			if ( $total_results === 1 ) {
+				printf( esc_html__( 'Showing 1 of 1 result', 'shiro' ) );
```

Comment:
I don't see it being applied here, as the message is for 
- No results, no message
- single result
- more than one result

But I will be available for any opposite opinions.

Reply:
Yeah, it doesn't make a difference logically, it's just a nice function for translation, especially when you're translating into languages that have difference rules for pluralization. In this case, it'd give you something like

```
printf(
    _n(
        'Showing %1$d of %3$d result',
        'Showing %1$d&ndash;%2$d of %3$d results',
        $total_results,
        'shiro'
    ),
    $first_result,
    $last_result,
    $total_results
);

```

Ultimately, like you mentioned on standup, all of this is academic because WIkimedia doesn't use a standard translation system. But if they were, this has the benefit of allowing languages that pluralize differently to be translated naturally. (As one example, in Russian there are 3 different plural cases for nouns - one for numbers ending in 1, one for numbers ending in 2-4, and another for numbers ending in 5-0.)